src/rpc/blockchain.cpp

```diff
@@ -1179,7 +1182,23 @@ UniValue getblockchaininfo(const JSONRPCRequest& request)
     obj.push_back(Pair("mediantime",            (int64_t)chainActive.Tip()->GetMedianTimePast()));
     obj.push_back(Pair("verificationprogress",  GuessVerificationProgress(Params().TxData(), chainActive.Tip())));
     obj.push_back(Pair("chainwork",             chainActive.Tip()->nChainWork.GetHex()));
+    obj.push_back(Pair("size_on_disk",          CalculateCurrentUsage()));
     obj.push_back(Pair("pruned",                fPruneMode));
+    if (fPruneMode) {
+        CBlockIndex* block = chainActive.Tip();
+        assert(block);
+        while (block->pprev && (block->pprev->nStatus & BLOCK_HAVE_DATA)) {
+            block = block->pprev;
+        }
+
+        obj.push_back(Pair("pruneheight",        block->nHeight));
+
+        bool fPruneAuto = (gArgs.GetArg("-prune", 0) > 1);
```

Comment:
current code style is snake_case for variables (https://github.com/bitcoin/bitcoin/blob/master/doc/developer-notes.md#developer-notes). `prune_auto` or `auto_prune` should be fine.

Reply:
nit: Since `1` is a magic number meaning manual pruning, I think it's clearer to use equality like in init.cpp, rather than greater-than:

```
        bool fPruneAuto = (gArgs.GetArg("-prune", 0) != 1);
```

Reply:
I was torn there, because that actually gives True in the case prune=0, but that case can never happen because you'd skip the if (fPruneMode) block. I'll change it.